internal/pkg/utils/utils_go118.go

```diff
@@ -0,0 +1,23 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.18
+// +build go1.18
+
+package utils
+
+func init() {
+	// After 1.18 default rendering of interface{} is any.
+	DefaultEmptyInterface = "any"
+}
```

Comment:
(a) This is a clever, clean way to do this sort of versioning that I haven't had the occasion to see before.  Neat!

(b) I can see it going either way, but would it be better to just keep it simple and bump the `go.mod` to require 1.18?  I don't see us maintaining distinct branches for backwards compatibility, and the old version is tagged if someone using the scanner is working at pre-1.18.

Then again, it's not _that_ much mental overhead here, so keeping it as is and more versatile could be good, unless / until there's any pressure to simplify it in the future.

Reply:
Hmmm, related, I see that I've let #325 go stale without review, and that _does_ bump to go1.18.

@guodongli-google, if this PR is merged, do you see that throwing a huge wrench in #325?  Would that break without the go1.18 bump?

Reply:
I'll try to summarize the delta between this and #325. This is doing the minimum to be backwards compatible, fixes tests, bumps only the versions it needs to (CI env testing requirements) and prevents crashes while not changing the solvers. #325 bumps the version to 1.18 for simplicity, and does minimal changes to the solvers to prevent flows through type parameters and suppresses complaining (false negatives). This does not suppress type parameter complaining. So there is a lot of overlap between these two.

Whether to be backwards compatible or keep it simple and version bump is a judgement call for you :-)

Reply:
Gotcha.  This PR doesn't stop the `unexpected type received: *types.TypeParam T; please report this issue`, but it wasn't trying to.  That was my disconnect.